server/blob/blob-storage-strategy/src/main/java/org/apache/james/server/blob/deduplication/MinIOGenerationAwareBlobId.java

```diff
@@ -0,0 +1,160 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.server.blob.deduplication;
+
+import static org.apache.james.server.blob.deduplication.GenerationAwareBlobId.NO_FAMILY;
+import static org.apache.james.server.blob.deduplication.GenerationAwareBlobId.computeGeneration;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.util.Objects;
+
+import jakarta.inject.Inject;
+
+import org.apache.james.blob.api.BlobId;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Preconditions;
+
+public class MinIOGenerationAwareBlobId implements BlobId, GenerationAware {
+    public static class Factory implements BlobId.Factory {
+        public static final int NO_FAMILY = 0;
+        public static final int NO_GENERATION = 0;
+
+        private final Clock clock;
+        private final GenerationAwareBlobId.Configuration configuration;
+        private final BlobId.Factory delegate;
+
+        @Inject
+        public Factory(Clock clock, GenerationAwareBlobId.Configuration configuration, BlobId.Factory delegate) {
+            this.clock = clock;
+            this.configuration = configuration;
+            this.delegate = delegate;
+        }
+
+        @Override
+        public BlobId parse(String id) {
+            int separatorIndex1 = id.indexOf('/');
```

Comment:
An alternative approach is to use the "family" character to identify the blobId: 
- family = 0 ->  delegate 
- family = 1 -> GenerationAwareBlobId 
- family = 2 -> MinIOGenerationAwareBlobId.

Reply:
TBH less flexible: what if I already incremented several time the family?
Which value to use as a discriminant?